### PR TITLE
math optimization

### PR DESCRIPTION
**Pull request by NouDaimon**

### CODE REVIEW

File: test/utils/Math.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { MathMock, MathMock__factory } from '../../typechain';
+
+describe('MathMock', function () {
```
[the remark this answers is not in the document]
Comment (NouDaimon) [reply test/utils/Math.ts:5]: Done

File: test/utils/Math.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        expect(
+          await instance.average(
+            ethers.BigNumber.from('6'),
+            ethers.BigNumber.from('5'),
+          ),
+        ).to.equal(ethers.BigNumber.from('5'));
+      });
+    });
+
+    describe('__internal sqrt', function () {
```
[the remark this answers is not in the document]
Comment (NouDaimon) [reply test/utils/Math.ts:32]: Done

File: test/utils/Math.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+describe('MathMock', function () {
+  let instance: MathMock;
+
+  before(async function () {
+    const [deployer] = await ethers.getSigners();
+    instance = await new MathMock__factory(deployer).deploy();
+  });
+
+  describe('__internal', function () {
+    describe('average', function () {
```
[the remark this answers is not in the document]
Comment (NouDaimon) [reply test/utils/Math.ts:14]: Done

File: contracts/utils/MathMock.sol [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+contract MathMock {
+    using Math for uint256;
+
+    function average(uint256 _a, uint256 _b) external pure returns (uint256) {
+        return _a.average(_b);
+    }
+
+    function sqrt(uint256 _x) external pure returns (uint256) {
+        return _x.sqrt();
```
Comment (ItsNickBarry) [inline contracts/utils/MathMock.sol:15]: Don't use underscore in variable names.  That convention is useful for distinguishing between storage variables and function arguments, but the SolidState library does not use storage variables anywhere.

File: test/utils/Math.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          await instance.average(
+            ethers.constants.MaxUint256,
+            ethers.constants.MaxUint256,
+          ),
+        ).to.equal(ethers.constants.MaxUint256);
+      });
+    });
+
+    describe('#sqrt', function () {
+      it('returns the sqrt of a positive integer from 0 to maxUint256', async function () {
```
Comment (ItsNickBarry) [inline test/utils/Math.ts:47]: Add a test for max value (MaxUint256 - 1).

**Review by ItsNickBarry — CHANGES_REQUESTED** [review]

File: contracts/utils/MathMock.sol (hunk 4 above)
Comment (NouDaimon) [reply contracts/utils/MathMock.sol:15]: I thought it was to distinguish between function arguments and any other variable, such as those initialized in the function body. 

Noted. Thank you.

File: test/utils/Math.ts (hunk 5 above)
Comment (NouDaimon) [reply test/utils/Math.ts:47]: Done.

**Pull request merged**